src/exceptiongroup/_catch.py

```diff
@@ -38,34 +37,32 @@ def __exit__(
 
         return False
 
-    def handle_exception(self, exc: BaseException) -> BaseException | None:
+    def handle_exception(self, exc: BaseException) -> BaseExceptionGroup | None:
+        excgroup: BaseExceptionGroup | None
         if isinstance(exc, BaseExceptionGroup):
-            matched, unmatched = exc.split(self._exc_types)
-            matched_exceptions = matched.exceptions if matched else ()
-            unhandled_exceptions: list[BaseException] = [unmatched] if unmatched else []
-
-            # Match the exceptions against the given type(s) and call this method to
-            # handle the individual exceptions
-            for matched_exc in matched_exceptions:
-                unhandled_exc = self.handle_exception(matched_exc)
-                if unhandled_exc is not None:
-                    unhandled_exceptions.append(unhandled_exc)
-
-            if unhandled_exceptions:
-                return BaseExceptionGroup("", unhandled_exceptions)
-            else:
-                return None
-
-        handler = self._handler_map.get(type(exc))
-        if handler is not None:
-            try:
-                handler(exc)
-            except BaseException as new_exc:
-                return new_exc
-            else:
-                return None
+            excgroup = exc
+        else:
+            excgroup = BaseExceptionGroup("", [exc])
+
+        new_exceptions: list[BaseException] = []
+        for exc_types, handler in self._handler_map.items():
+            matched, excgroup = excgroup.split(exc_types)
+            if matched:
+                try:
+                    handler(matched)
+                except BaseException as new_exc:
+                    new_exceptions.append(new_exc)
+
+            if not excgroup:
+                break
+
+        if new_exceptions:
+            if excgroup:
+                new_exceptions.append(excgroup)
+
+            return BaseExceptionGroup("", new_exceptions)
         else:
-            return exc
+            return excgroup
```

Comment:
This looks like it will wrap an incoming single exception in an ExceptionGroup if it doesn't match any of the handlers, which is different from the `except*` behavior.

Reply:
I've fixed this in 1.0.0rc7.